### PR TITLE
Fix the CI

### PR DESCRIPTION
**Pull request by nabobalis**
Trying to fix everything for the release of numpy 2.1

Need to see where else this errors. 

Fixes https://github.com/sunpy/sunpy/issues/7715

### CODE REVIEW

File: sunpy/timeseries/sources/goes.py [hunk 1]
```diff
@@ -228,12 +227,12 @@ def _parse_netcdf(filepath):
                 raise ValueError(f"No flux data (either a_flux or xrsa_flux) found in file: {filepath}")
             flux_name_a = flux_name.name
             flux_name_b = flux_name_a.replace("a", "b")
-            xrsa = np.array(h5nc[flux_name_a])
-            xrsb = np.array(h5nc[flux_name_b])
+            xrsa = np.asarray(h5nc[flux_name_a])
```
Comment (nabobalis) [inline sunpy/timeseries/sources/goes.py:230]: numpy only wants to accept array inputs that implement copy and dtype. So switching to asarray bypasses this warning.

File: sunpy/timeseries/tests/test_timeseries_factory.py [hunk 2]
```diff
@@ -99,7 +98,6 @@ def test_from_url():
     assert isinstance(ts[1], sunpy.timeseries.GenericTimeSeries)
 
 
-@skip_numpy2
 def test_read_cdf():
```
Comment (nabobalis) [inline sunpy/timeseries/tests/test_timeseries_factory.py:101]: This now passes with the updated cdflib.

**Comment by nabobalis** [comment]
Linkcheck is due to VSO

Conda is due to old cdflib package, the feedstock repo needs some love. I don't see the issue with merging this PR otherwise.

File: sunpy/net/vso/tests/test_vso.py [hunk 3]
```diff
@@ -322,7 +322,9 @@ def test_fallback_if_cgi_offline(mocker):
     with pytest.warns(SunpyConnectionWarning,
                       match=f"Connection to {cgi_url} failed with error .* Retrying with different url and port"):
         mirror = get_online_vso_url()
-    assert mirror["url"] != "http://docs.virtualsolar.org/WSDL/VSOi_rpc_literal.wsdl"
+    # This guards us if the fallback URL is also down
```
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/net/vso/tests/test_vso.py:325]: Fair. I will add that.

**Review by wtbarnes — APPROVED** [review]

File: sunpy/net/vso/tests/test_vso.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (nabobalis) [reply sunpy/net/vso/tests/test_vso.py]: I have patched this.

**Review by Cadair — APPROVED** [review]

**Pull request merged**